otherlibs/stdune/src/fpath.mli

```diff
@@ -5,12 +5,14 @@ type mkdir_result =
   | Created (** The directory was created. *)
   | Missing_parent_directory
   (** No parent directory, use [mkdir_p] if you want to create it too. *)
+  | Permission_denied (** Could not create the directory *)
```

Comment:
I check our code internally, and it turns that we have many uses of this function that would need to be modified. Instead of modifying this function, can you add a new function such as `val mkdir_p_with_permissions : (mkdir_p_result, [`Permission_denied]) result` and use that? That would make syncing a lot simpler for us.